account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -246,7 +246,7 @@ def _pull(self, date_since, date_until):
                                 )
                             )
                         continue
-                    elif date <= date_since or date > date_until:
+                    elif date <= statement_date_since or date > statement_date_until:
```

Comment:
These cases are covered by clauses above, and this change breaks the logic due to `<` vs `<=` and `>` vs `>=`

Reply:
in the wizard if you set 01/01 to 01/02, if you have in monthly statement:

`_obtain_statement_data(statement_date_since, statement_date_until)`
it upload transaction 01/01 to 01/02
then it upload transaction 01/01 to **01/03**

if you keep your filter `elif date <= date_since or date > date_until:` all transaction between 02/01 to 28/02 are removed, and you balance are false.

that why it is better to use `elif date <= statement_date_since or date > statement_date_until:`

no ?

Reply:
If you have monthly statement provider, and use wizard to pull Jan 1 to Feb 1, it will pull
Jan 1 - Jan 31 as one statement and Feb 1 - Feb 1 as another statement, that is by design. As you've requested only until Feb 1st, the monthly/weekly/daily selection affects on how transactions are split between statements.

Reply:
On top of that, `date < statement_date_since` is handled properly 20 lines above the check that you refer to

Reply:
ok, but the issues is it create an empty statement

Reply:
It's not an issue, since for requested date statement IS empty

Reply:
When/if that statement will be having any line, it will become not empty. On top of that, statement date is the beginning of the period that is covered by statement, so it HAS to be created.

Reply:
I have modified the commit, now it create an empty bank statement with the good balance.

Reply:
What do you mean by good balance? And what is has to do with what we've discussed above?